cmd/minishift/cmd/util/oc.go

```diff
@@ -92,3 +87,32 @@ func SetOcContext(profileName string) error {
 
 	return nil
 }
+
+//RemoveCurrentContext removes the current context from `machinename_kubeconfig`
+func RemoveCurrentContext() error {
+	ocPath := minishiftConfig.InstanceConfig.OcPath
+	cmd := "config unset current-context"
+
+	ocRunner, err := oc.NewOcRunner(ocPath, filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig"))
+	if err != nil {
+		if glog.V(2) {
+			fmt.Println(fmt.Sprintf("%s", err.Error()))
+		}
+		return errors.New(fmt.Sprintf("Error unsetting current-context"))
+	}
+
+	exitCode := ocRunner.RunAsUser(cmd, nil, nil)
+	if exitCode != 0 {
+		return errors.New(fmt.Sprintf("Error during removing current context: %v", err.Error()))
+	}
+	return nil
+}
+
+func GetOcPath(profileName string) string {
+	instanceConfigFile := filepath.Join(constants.GetProfileHomeDir(), "machines", profileName+".json")
+	instanceConfig, err := minishiftConfig.NewInstanceConfig(instanceConfigFile)
```

Comment:
the function `GetOcPath` loads the `machinename.json` file and reads the `OcPath` from there based on the profilename, if the VM exists this file will be there, if the VM doesn't exist we don't need it anyway. Also, for unsetting the current context we can get the `OcPath` from `instanceConfig.OcPath` only while setting the context we need to call `GetOcPath` because then we need the `OcPath` of the profile we are switching context to, for unsetting we use the `OcPath` of the active profile.